cloudinit/config/cc_reset_rmc.py

```diff
@@ -0,0 +1,135 @@
+# (c) Copyright IBM Corp. 2020 All Rights Reserved
+#
+# Author: Aman Kumar Sinha <amansi26@in.ibm.com>
+#
+# This file is part of cloud-init. See LICENSE file for license information.
+
+
+"""
+Reset RMC
+------------
+**Summary:** reset rsct node id
+
+Reset RMC module is IBM PowerVM Hypervisor specific
+
+Reliable Scalable Cluster Technology (RSCT) is a set of software components,
+that  together provide a comprehensive clustering environment (RAS features)
+for IBM PowerVM based virtual machines. RSCT includes the Resource monitoring
+and control (RMC) subsystem. RMC is a generalized framework used for managing,
+monitoring, and manipulating resources. RMC runs as a daemon process on
+individual machines and needs creation of unique node id and restarts
+during VM boot.
+More details refer
+https://www.ibm.com/support/knowledgecenter/en/SGVKBA_3.2/admin/bl503_ovrv.htm
+
+This module handles
+- creation of the unique RSCT node id to every instance/virtual machine
+  and ensure once set, it isn't changed subsequently by cloud-init.
+  In order to do so, it restarts RSCT service.
+
+Prerequisite of using this module is to install RSCT packages.
+
+**Internal name:** per instance
+
+**Supported distros:** rhel, sles and ubuntu
+
+"""
+import os
+
+from cloudinit import log as logging
+from cloudinit.settings import PER_INSTANCE
+from cloudinit import util
+from cloudinit import subp
+
+frequency = PER_INSTANCE
+PIDOF = 'pidof'
+
+# RMCCTRL is expected to be in system PATH (/opt/rsct/bin)
+# The symlink for RMCCTRL and RECFGCT are
+# /usr/sbin/rsct/bin/rmcctrl and
+# /usr/sbin/rsct/install/bin/recfgct respectively.
+RSCT_PATH = '/opt/rsct/install/bin'
+RMCCTRL = 'rmcctrl'
+RECFGCT = 'recfgct'
+
+LOG = logging.getLogger(__name__)
+
+NODE_ID_FILE = '/etc/ct_node_id'
+
+
+def handle(name, _cfg, cloud, _log, _args):
+    # Ensuring node id has to be generated only once during first boot
+    if cloud.datasource.platform_type == 'none':
+        LOG.debug('Skipping creation of new ct_node_id node')
+        return
+
+    orig_path = add_path()
+
+    LOG.debug('Attempting to reset RMC.')
+
+    node_id_before = get_node_id()
```

Comment:
I do think its kind of odd that we are checking that the node_id changes.
As far as I can tell, all we're doing is 
 * reading a file
 * calling RECFGCT (and checking its return code).
 * reading the file again

Is it true that If RECFGCT returns zero (success), then the node_id should be updated?
It seems an arbitrary thing for us to pick as an extra measurement of success.  
Is there a valid case where RECFGCT would exit success and not update NODE_ID_FILE?

If you've hit that situation, then I would suggest filing a bug against the application.

Its fine to leave this in if you feel the log messages are useful, but.... RECFGCT really should exit non-zero if it fails.

Reply:
As per my understanding, RECFGCT exit success, it update NODE_ID.  I would like to keep these changes. if that is ok with the  you. 

If in case we hit the issue you mention, the bug should be on the third party library (RSCT) not on cloud init . I agree with this

Reply:
> As per my understanding, RECFGCT exit success, it update NODE_ID. I would like to keep these changes. if that is ok with the you.
> 
> If in case we hit the issue you mention, the bug should be on the third party library (RSCT) not on cloud init . I agree with this

You can keep them.  It just seems arbitrary.  That program could exit success and not have intended side effects (of a service running) in any number of ways... its odd to pick this one to pay attention to specifically.